### PR TITLE
update cli release workflow

### PR DESCRIPTION
**Pull request by 0xjinn**
### Description

Update CLI Release work flow

1. updated `release_tag` to `release_version`
2. Use formatting on actual release tag and title
3. Add `dry_run` boolean to allow dry_run instead of actual release

### CODE REVIEW

**Comment by 0xjinn** [comment]
@rustielin First time updating github workflow. Wanted to make sure what I am doing is correct 😄 

What is the best way to test this and visual the UI before merging/commiting?

File: .github/workflows/cli-release.yaml [hunk 1]
```diff
@@ -93,6 +94,7 @@ jobs:
     permissions:
       contents: "write"
       pull-requests: "read"
+    if: ${{ github.event.inputs.dry_run }} == 'true'
```
Comment (rustielin) [inline .github/workflows/cli-release.yaml:97]: `if: ${{ inputs.dry_run }}` should work

File: .github/workflows/cli-release.yaml [hunk 2]
```diff
@@ -102,8 +104,8 @@ jobs:
         uses: marvinpinto/action-automatic-releases@919008cf3f741b179569b7a6fb4d8860689ab7f0 # pin@v1.2.1
         with:
           repo_token: "${{ secrets.GITHUB_TOKEN }}"
-          automatic_release_tag: "${{ github.event.inputs.release_tag }}"
+          automatic_release_tag: "${{ format('aptos-cli-v{0}', github.event.inputs.release_version) }}"
```
Comment (rustielin) [inline .github/workflows/cli-release.yaml:107]: same with `inputs.release_version`

**Review by rustielin — COMMENTED** [review]
iiuc `github.event.inputs.*` is the same as `inputs.*1

**Review by rustielin — COMMENTED** [review]
image


You can also test by workflow_dispatch. https://github.com/aptos-labs/aptos-core/actions/workflows/cli-release.yaml

Make sure to set dry_run

File: .github/workflows/cli-release.yaml [hunk 3]
```diff
@@ -93,6 +94,7 @@ jobs:
     permissions:
       contents: "write"
       pull-requests: "read"
+    if: ${{ inputs.dry_run }} == 'true'
```
Comment (gregnazario) [inline .github/workflows/cli-release.yaml:97]: False?

**Review by gedigi — APPROVED** [review]

File: .github/workflows/cli-release.yaml (hunk 3 above)
Comment (0xjinn) [reply .github/workflows/cli-release.yaml]: Good catch!

**Review by gregnazario — APPROVED** [review]

**Pull request merged**